java/ql/lib/semmle/code/java/security/Files.qll

```diff
@@ -0,0 +1,53 @@
+/** Provides classes and predicates for working with file objects. */
+
+import java
+import semmle.code.java.dataflow.ExternalFlow
+
+private class CreateFileSinkModels extends SinkModelCsv {
+  override predicate row(string row) {
+    row =
+      [
+        "java.io;FileOutputStream;false;FileOutputStream;;;Argument[0];create-file",
+        "java.io;RandomAccessFile;false;RandomAccessFile;;;Argument[0];create-file",
+        "java.io;FileWriter;false;FileWriter;;;Argument[0];create-file",
+        "java.io;PrintWriter;false;PrintWriter;(File);;Argument[0];create-file",
+        "java.io;PrintWriter;false;PrintWriter;(File,String);;Argument[0];create-file",
+        "java.io;PrintWriter;false;PrintWriter;(String);;Argument[0];create-file",
+        "java.io;PrintWriter;false;PrintWriter;(String,String);;Argument[0];create-file",
+        "java.nio.file;Files;false;move;;;Argument[1];create-file",
+        "java.nio.file;Files;false;copy;;;Argument[1];create-file",
+        "java.nio.file;Files;false;newOutputStream;;;Argument[0];create-file",
+        "java.nio.file;Files;false;newBufferedReader;;;Argument[0];create-file",
```

Comment:
Yes, it's not possible to select sinks based on certain argument values with the current CSV models AFAIK. But it's something to keep in mind because that's in fact an additional sink we should take into account.

Reply:
Would it make sense to add a comment in the row where that method would be, explaining that it can currently not be modelled?